src/shared/loopback-setup.c

```diff
@@ -114,9 +114,15 @@ static int add_ipv6_address(sd_netlink *rtnl, struct state *s) {
         if (r < 0)
                 return r;
 
-        r = sd_rtnl_message_addr_set_flags(req, IFA_F_PERMANENT);
+        uint32_t flags = IFA_F_PERMANENT|IFA_F_NOPREFIXROUTE;
+        r = sd_rtnl_message_addr_set_flags(req, flags & 0xffu);
         if (r < 0)
                 return r;
+        if ((flags & ~0xffu) != 0) {
```

Comment:
this fallback is not very clear either, a short comment explaining why it's needed would be nice